infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -517,18 +537,35 @@ public void updateTopology(LogReplicationClusterInfo.TopologyConfigurationMsg to
      * msg needs to send out.
      */
     @Override
-    public void prepareSiteRoleChange() {
-        replicationManager.prepareSiteRoleChange();
+    public void prepareClusterRoleChange() {
```

Comment:
I see we have a processClusterRoleChange which comes as part of a DISCOVERED_TOPOLOGY event, but who can directly act upon the discovery service to forcefully prepare for Cluster Role Change? Is this some testing API? we should mark it as such and document it if it's the case.

Reply:
This is an API we talked about that the upper layer like site manager can call us to prepare a site role change. Now seems that it may not be required. We can get rid of it or just keep it for our testing or warning purpose. The goal of this API is that after this call, the tables that should be replicated should not be changed any more. This hook can tell if the data has been changed or not.